src/webui/api/serialize/serialize_torrent.cpp

```diff
@@ -85,6 +87,39 @@ namespace
             return u"unknown"_s;
         }
     }
+
+    // NOTE: keep logic from trackersfilterwidget.cpp in sync with these methods
+    // TODO: ^^ make an indirection to avoid duplicate logic
+    bool torrentHasTrackerError(const QList<BitTorrent::TrackerEntryStatus> tstatuses) {
+        return std::any_of(
+            tstatuses.begin(), tstatuses.end(),
+            [](const BitTorrent::TrackerEntryStatus &tstatus) {
+                return tstatus.state == BitTorrent::TrackerEndpointState::TrackerError;
+            }
+        );
+    }
+    bool torrentHasAnnounceError(const QList<BitTorrent::TrackerEntryStatus> tstatuses) {
+        return std::any_of(
+            tstatuses.begin(), tstatuses.end(),
+            [](const BitTorrent::TrackerEntryStatus &tstatus) {
+                return (tstatus.state == BitTorrent::TrackerEndpointState::NotWorking)
+                        || (tstatus.state == BitTorrent::TrackerEndpointState::Unreachable);
+            }
+        );
+    }
+    bool torrentHasWarning(const QList<BitTorrent::TrackerEntryStatus> tstatuses) {
+        return std::any_of(
+            tstatuses.begin(), tstatuses.end(),
+            [](const BitTorrent::TrackerEntryStatus &tstatus) {
+                return std::any_of(
+                    tstatus.endpoints.cbegin(), tstatus.endpoints.cend(), 
+                    [](const BitTorrent::TrackerEndpointStatus &endpointEntry) {
+                        return !endpointEntry.message.isEmpty() && endpointEntry.state == BitTorrent::TrackerEndpointState::Working;
+                    }
+                );
+            }
+        );
+    }
```

Comment:
A total violation of qBittorrent coding style.

Reply:
is now ok?